### PR TITLE
Update NOTICE.txt

### PR DESCRIPTION
**Pull request by amyblais**
See comments below.

Also I wasn't sure if the server repo has any updates for the NOTICE.txt.

### CODE REVIEW

File: NOTICE.txt [hunk 1]
```diff
@@ -7,6 +7,20 @@ NOTICES:
 This document includes a list of open source components used in Mattermost Web App, including those that have been modified.
 
 --------
+
+## @types/enzyme
```
Comment (amyblais) [inline NOTICE.txt]: This commit has many ``@types``: https://github.com/mattermost/mattermost-webapp/commit/09d9fc3a00b176831e4cf3c3ceef4d25bde4dc14#diff-b9cfc7f2cdf78a7f4b91a753d10865a2. Is there a general one for all of them?

File: NOTICE.txt [hunk 2]
```diff
@@ -7,6 +7,20 @@ NOTICES:
...
+
+This product contains 'regenerator-runtime' by XuCat.
+
+For WeChat using async/await
+
+* HOMEPAGE:
+  * https://github.com/xubaifuCode/regeneratorRuntime
+
+* LICENSE:
```
Comment (amyblais) [inline NOTICE.txt]: This doesn't have a license. Also, I'm not sure if this is the right one.

File: NOTICE.txt [hunk 3]
```diff
@@ -2601,6 +2615,19 @@ SOFTWARE.
...
+## regenerator-runtime
+
+This product contains 'regenerator-runtime' by XuCat.
+
+For WeChat using async/await
+
+* HOMEPAGE:
+  * https://github.com/xubaifuCode/regeneratorRuntime
+
+* LICENSE:
```
Comment (amyblais) [inline NOTICE.txt]: Doesn't have a license.

File: NOTICE.txt [hunk 4]
```diff
@@ -2636,6 +2663,19 @@ SOFTWARE.
...
+## semver
+
+This product contains 'semver' by semver.
+
+Semantic Versioning Specification
+
+* HOMEPAGE:
+  * https://github.com/semver/semver
+
+* LICENSE:
```
Comment (amyblais) [inline NOTICE.txt]: Doesn't have a license.

File: NOTICE.txt [hunk 5]
```diff
@@ -2672,6 +2712,73 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 ---
 
+## typescript
+
+This product contains 'typescript' by Microsoft.
```
Comment (amyblais) [inline NOTICE.txt]: Not 100% sure if this is the right one.

File: NOTICE.txt (hunk 3 above)
Comment (crspeller) [reply NOTICE.txt]: Looks like they come from here: https://github.com/DefinitelyTyped/DefinitelyTyped

File: NOTICE.txt (hunk 1 above)
Comment (crspeller) [reply NOTICE.txt]: Looks like they come from here: https://github.com/DefinitelyTyped/DefinitelyTyped

File: NOTICE.txt (hunk 3 above)
Comment (crspeller) [reply NOTICE.txt]: It's this one: https://github.com/facebook/regenerator

File: NOTICE.txt (hunk 4 above)
Comment (crspeller) [reply NOTICE.txt]: Where are you getting this from? I don't see it.

File: NOTICE.txt (hunk 5 above)
Comment (crspeller) [reply NOTICE.txt]: Looks right to me.

File: NOTICE.txt (hunk 4 above)
Comment (amyblais) [reply NOTICE.txt]: I got it from here: https://github.com/mattermost/mattermost-webapp/commit/e709f99ccde988bd9566f56b7fa6f8a6253444f5#diff-b9cfc7f2cdf78a7f4b91a753d10865a2.

But I don't see it in package.json anymore either so maybe it was removed.

Comment (amyblais) [reply NOTICE.txt]: @crspeller It is being re-added to v5.16 with this PR: https://github.com/mattermost/mattermost-webapp/pull/3807.

Comment (crspeller) [reply NOTICE.txt]: Alright. But if it doesn't have a license I don't think we can use it. The default is all rights reserved.

Comment (deanwhillier) [reply NOTICE.txt]: Hey @amyblais, the package URL is actually `https://github.com/npm/node-semver` and it's license [is here](https://github.com/npm/node-semver/blob/master/LICENSE).

Comment (amyblais) [reply NOTICE.txt]: Thank you :)

**Comment by amyblais** [comment]
@crspeller This is good for review.

**Comment by amyblais** [comment]
Chris, hope you can also help merge and cherry-pick this.

**Pull request merged**